raiden/tests/unit/channel_state/test_coop_settle.py

```diff
@@ -0,0 +1,448 @@
+    partner_model1, privkey2 = create_model(balance=100)
+    signer = LocalSigner(privkey2)
+    channel_state = create_channel_from_models(our_model1, partner_model1, privkey2)
+    block_hash = make_block_hash()
+    block_number = 1
+    expiration = 1000
+
+    withdraw_request = _make_receive_coop_settle_withdraw_request(
+        channel_state,
+        pseudo_random_generator,
+        signer,
+        nonce=1,
+        total_withdraw=100,
+        withdraw_expiration=expiration,
+    )
+
+    iteration = channel.state_transition(
+        channel_state=channel_state,
+        state_change=withdraw_request,
+        block_hash=block_hash,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=block_number,
+    )
+
+    # Don't accept incoming coop-settle withdraw requests when there are pending
+    # transfers
+    assert (
+        search_for_item(
+            iteration.events, EventInvalidReceivedWithdrawRequest, {"attempted_withdraw": 100}
+        )
+        is not None
+    )
+
+
+def test_receive_request():
+    pseudo_random_generator = random.Random()
+
+    our_model1, _ = create_model(balance=70)
+    partner_model1, privkey2 = create_model(balance=100)
+    signer = LocalSigner(privkey2)
+    channel_state = create_channel_from_models(our_model1, partner_model1, privkey2)
+    block_hash = make_block_hash()
+    block_number = 1
+    expiration = 1000
+
+    withdraw_request = _make_receive_coop_settle_withdraw_request(
+        channel_state,
+        pseudo_random_generator,
+        signer,
+        nonce=1,
+        total_withdraw=100,
+        withdraw_expiration=expiration,
+    )
+    iteration = channel.state_transition(
+        channel_state=channel_state,
+        state_change=withdraw_request,
+        block_hash=block_hash,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=block_number,
+    )
+
+    assert (
+        search_for_item(
+            iteration.events,
+            SendWithdrawConfirmation,
+            {"total_withdraw": 100, "nonce": 2, "expiration": expiration},
+        )
+        is not None
+    )
+    assert (
+        search_for_item(
+            iteration.events,
+            SendWithdrawRequest,
+            {"total_withdraw": 70, "nonce": 1, "coop_settle": False, "expiration": expiration},
+        )
+        is not None
+    )
+
+    # our view
+    _assert_coop_settle_state(
+        channel_state.partner_state,
+        channel_state.our_state,
+        our_initial_model=partner_model1,
+        our_current_nonce=1,
+        coop_settle_expiration=expiration,
+        initiated_coop_settle=True,
+    )
+    # partner's view
+    _assert_coop_settle_state(
+        channel_state.our_state,
+        channel_state.partner_state,
+        our_initial_model=our_model1,
+        our_current_nonce=2,
+        coop_settle_expiration=expiration,
+        initiated_coop_settle=False,
+    )
+
```

Comment:
The comments here need to be swapped -- "our view" should be where "partner's view" is and vice versa.

Reply:
Done.